src/shell/dotnet/Shell/App.xaml.cs

```diff
@@ -113,8 +112,20 @@ private void ConfigureServices(HostBuilderContext context, IServiceCollection se
                 .UseServer()
                 .UseAccessToken(_messageRouterAccessToken));
 
+        //This can be replaced by the `InjectFdc3BackendServiceIfEnabledFromConfig` extension method
+        ///*
+        var fdc3Configuration = context.Configuration
+            .GetSection(Fdc3Options.Fdc3OptionsName);
+
+        var fdc3Options = fdc3Configuration
+            .Get<Fdc3Options>();
 
-        services.AddFdc3DesktopAgent(fdc3 => fdc3.WithUserChannel("default"));
+        if (fdc3Options != null && fdc3Options.EnableFdc3)
+        {
+            services.Configure<Fdc3Options>(fdc3Configuration);
+            services.AddFdc3DesktopAgent(builder => builder.WithUserChannel(fdc3Options.ChannelId ?? "default"));
```

Comment:
I'd remove `WithUserChannel` and just add a `Configure<Fdc3Options>` step that applies this default.